### ISSUE TITLE
sqlj.install_jar() failure with LANG=de_DE.UTF-8: FEHLER:  java.sql.SQLException: Exception verifying variable-length data: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 22; cvc-elt.1.a: Deklaration des Elements "pljava-content-wrap" kann nicht gefunden werden.

### ISSUE THREAD

**Issue by df7cb** [tada/pljava#312]
Hi,

with pljava 1.5.6, sqlj.install_jar() fails if LANG is de_DE.UTF-8. The error message is 

`FEHLER:  java.sql.SQLException: Exception verifying variable-length data: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 22; cvc-elt.1.a: Deklaration des Elements "pljava-content-wrap" kann nicht gefunden werden.`

which translates to `declaration of element "pljava-content-wrap" could not be found`.

Good behavior with LANG=C.UTF-8 (and en_US.UTF-8): (Some PG-side messages are still German, but whatever)
```
+ psql -c 'SELECT sqlj.install_jar('\''file:/home/cbe/projects/postgresql/pljava/postgresql-pljava/debian/postgresql-13-pljava/usr/share/postgresql/13/pljava/pljava-examples-1.5.6.jar'\'', '\''examples'\'', true)'
INFO:  05 Oct 20 10:24:39 org.postgresql.pljava.example.LoggerTest Looking good!
INFO:  05 Oct 20 10:24:39 org.postgresql.pljava.example.annotation.SPIActions issue 228 tests ok
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest Holdability OK
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.management.Commands Type id = OID(600)
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.annotation.Point pg_catalog.point from SQLInput
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.management.Commands Type id = OID(16650)
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.annotation.ComplexTuple javatest.complextuple from SQLInput
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.annotation.ComplexTuple javatest.complextuple to SQLOutput
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.annotation.ComplexTuple javatest.complextuple from SQLInput
NOTICE:  return type javatest.udtscalariotest is only a shell
NOTICE:  argument type javatest.udtscalariotest is only a shell
NOTICE:  return type javatest.udtscalariotest is only a shell
NOTICE:  argument type javatest.udtscalariotest is only a shell
NOTICE:  return type javatest.varlenaudttest is only a shell
NOTICE:  argument type javatest.varlenaudttest is only a shell
NOTICE:  return type javatest.varlenaudttest is only a shell
NOTICE:  argument type javatest.varlenaudttest is only a shell
NOTICE:  return type javatest.complex is only a shell
NOTICE:  argument type javatest.complex is only a shell
NOTICE:  return type javatest.complex is only a shell
NOTICE:  argument type javatest.complex is only a shell
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.annotation.ComplexScalar javatest.complex from string
ZEILE 1: SELECT javatest.assertHasValues( CAST('(1,2)' AS javatest.co...
                                               ^
ANFRAGE:  SELECT javatest.assertHasValues( CAST('(1,2)' AS javatest.complex), 1, 2)
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.annotation.ComplexScalar javatest.complex to SQLOutput
ZEILE 1: SELECT javatest.assertHasValues( CAST('(1,2)' AS javatest.co...
                                               ^
ANFRAGE:  SELECT javatest.assertHasValues( CAST('(1,2)' AS javatest.complex), 1, 2)
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.annotation.ComplexScalar javatest.complex from SQLInput
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest trigger transition table test ok
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest works for 32767
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest works for 32768
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest works for 65536
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest works for 1048576
NOTICE:  return type javatest.intwithmod is only a shell
NOTICE:  argument type javatest.intwithmod is only a shell
NOTICE:  return type javatest.intwithmod is only a shell
NOTICE:  argument type javatest.intwithmod is only a shell
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest SetOfRecordTest ok
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.annotation.PreJSR310 issue 199 test ok
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest timestamp roundtrip passes
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest timestamptz roundtrip passes
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.annotation.Point pg_catalog.point from SQLInput
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest issue192 test passes
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest boolean[] passes
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest "char"[] passes
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest bytea passes
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest int2[] passes
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest int4[] passes
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest int8[] passes
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest float4[] passes
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest float8[] passes
INFO:  05 Oct 20 10:24:40 org.postgresql.pljava.example.LoggerTest text[] passes
INFO:  05 Oct 20 10:24:42 org.postgresql.pljava.example.LoggerTest all Unicode codepoint ranges roundtripped successfully.
INFO:  05 Oct 20 10:24:42 org.postgresql.pljava.management.Commands Type id = OID(16732)
INFO:  05 Oct 20 10:24:45 org.postgresql.pljava.example.LoggerTest SQLXML echos succeeded
INFO:  05 Oct 20 10:24:46 org.postgresql.pljava.example.LoggerTest proxied SQLXML echos succeeded
INFO:  05 Oct 20 10:24:46 org.postgresql.pljava.example.LoggerTest XML Schema tests succeeded
INFO:  05 Oct 20 10:24:46 org.postgresql.pljava.example.LoggerTest java.time.LocalDate passes
INFO:  05 Oct 20 10:24:46 org.postgresql.pljava.example.LoggerTest java.time.LocalTime passes
INFO:  05 Oct 20 10:24:46 org.postgresql.pljava.example.LoggerTest java.time.OffsetTime passes
INFO:  05 Oct 20 10:24:46 org.postgresql.pljava.example.LoggerTest java.time.LocalDateTime passes
INFO:  05 Oct 20 10:24:46 org.postgresql.pljava.example.LoggerTest java.time.OffsetDateTime passes
INFO:  05 Oct 20 10:24:46 org.postgresql.pljava.example.LoggerTest OffsetTime as stmt param passes
 install_jar 
-------------
 
(1 Zeile)
```

Error with LANG=de_DE.UTF-8:

```
+ psql -c 'SELECT sqlj.install_jar('\''file:/home/cbe/projects/postgresql/pljava/postgresql-pljava/debian/postgresql-13-pljava/usr/share/postgresql/13/pljava/pljava-examples-1.5.6.jar'\'', '\''examples'\'', true)'
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest Looking good!
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.annotation.SPIActions issue 228 tests ok
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest Holdability OK
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.management.Commands Type id = OID(600)
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.annotation.Point pg_catalog.point from SQLInput
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.management.Commands Type id = OID(16650)
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.annotation.ComplexTuple javatest.complextuple from SQLInput
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.annotation.ComplexTuple javatest.complextuple to SQLOutput
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.annotation.ComplexTuple javatest.complextuple from SQLInput
HINWEIS:  Rückgabetyp javatest.udtscalariotest ist nur eine Hülle
HINWEIS:  Argumenttyp javatest.udtscalariotest ist nur eine Hülle
HINWEIS:  Rückgabetyp javatest.udtscalariotest ist nur eine Hülle
HINWEIS:  Argumenttyp javatest.udtscalariotest ist nur eine Hülle
HINWEIS:  Rückgabetyp javatest.varlenaudttest ist nur eine Hülle
HINWEIS:  Argumenttyp javatest.varlenaudttest ist nur eine Hülle
HINWEIS:  Rückgabetyp javatest.varlenaudttest ist nur eine Hülle
HINWEIS:  Argumenttyp javatest.varlenaudttest ist nur eine Hülle
HINWEIS:  Rückgabetyp javatest.complex ist nur eine Hülle
HINWEIS:  Argumenttyp javatest.complex ist nur eine Hülle
HINWEIS:  Rückgabetyp javatest.complex ist nur eine Hülle
HINWEIS:  Argumenttyp javatest.complex ist nur eine Hülle
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.annotation.ComplexScalar javatest.complex from string
ZEILE 1: SELECT javatest.assertHasValues( CAST('(1,2)' AS javatest.co...
                                               ^
ANFRAGE:  SELECT javatest.assertHasValues( CAST('(1,2)' AS javatest.complex), 1, 2)
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.annotation.ComplexScalar javatest.complex to SQLOutput
ZEILE 1: SELECT javatest.assertHasValues( CAST('(1,2)' AS javatest.co...
                                               ^
ANFRAGE:  SELECT javatest.assertHasValues( CAST('(1,2)' AS javatest.complex), 1, 2)
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.annotation.ComplexScalar javatest.complex from SQLInput
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest trigger transition table test ok
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest works for 32767
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest works for 32768
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest works for 65536
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest works for 1048576
HINWEIS:  Rückgabetyp javatest.intwithmod ist nur eine Hülle
HINWEIS:  Argumenttyp javatest.intwithmod ist nur eine Hülle
HINWEIS:  Rückgabetyp javatest.intwithmod ist nur eine Hülle
HINWEIS:  Argumenttyp javatest.intwithmod ist nur eine Hülle
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest SetOfRecordTest ok
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.annotation.PreJSR310 issue 199 test ok
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest timestamp roundtrip passes
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest timestamptz roundtrip passes
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.annotation.Point pg_catalog.point from SQLInput
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest issue192 test passes
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest boolean[] passes
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest "char"[] passes
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest bytea passes
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest int2[] passes
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest int4[] passes
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest int8[] passes
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest float4[] passes
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest float8[] passes
INFO:  05 Okt. 20 11:45:47 org.postgresql.pljava.example.LoggerTest text[] passes
INFO:  05 Okt. 20 11:45:50 org.postgresql.pljava.example.LoggerTest all Unicode codepoint ranges roundtripped successfully.
INFO:  05 Okt. 20 11:45:50 org.postgresql.pljava.management.Commands Type id = OID(16732)
INFO:  05 Okt. 20 11:45:53 org.postgresql.pljava.example.LoggerTest SQLXML echos succeeded
INFO:  05 Okt. 20 11:45:54 org.postgresql.pljava.example.LoggerTest proxied SQLXML echos succeeded
FEHLER:  java.sql.SQLException: Exception verifying variable-length data: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 22; cvc-elt.1.a: Deklaration des Elements "pljava-content-wrap" kann nicht gefunden werden.
```

Full (non-failing) logs at https://pgdgbuild.dus.dg-i.net/job/postgresql-pljava-binaries

(This is not a blocker for the Debian package as we aren't actually building or testing using this locale, but I figured I'd still report it.)

**Comment by jcflack** [comment]
Oh, for the love of Pete.

The regex being used to match that was already

    ^cvc-elt\.1(?:\.a)?+:.*'pljava-content-wrap'

but the German message uses `"` instead of `'`. Per Wikipedia, I suppose I'm lucky it wasn't `„…“` or `‚…‘` or `»…«` or `›…‹`


So I will have to relax that expression some. It should not be a blocker for anyone, so I think I will wait for a while to see what other issues get reported.

It should be possible for anyone who wants to install the examples to temporarily set the locale to `en_US.utf-8` just for the `install_jar`, and the examples will be successfully installed. Thereafter, it won't be possible (until this is fixed) to _use_ the XML Schema support supplied 'inside' (or „inside“) PL/Java's implementation, but the rest of the examples should work, and it remains possible to instantiate a `javax.xml.validation.Validator` or `ValidatorHandler` and apply it over PL/Java's API, as covered [in the docs][curtain].

[curtain]: https://tada.github.io/pljava/use/sqlxml.html#Pay_no_attention_to_that_man_behind_the_curtain

**Comment by df7cb** [comment]
As said, I don't think this needs an urgent fix. Glad you found the issue, though :)

There's various quotes used in German literature, but the more funky ones aren't really common in message strings. Most often it's "" or '', and rarely »«.

**Comment by jcflack** [comment]
Hi,

I have tested the `bug/master/issue312` branch here; with `LANG=de_DE.utf-8` in the environment I can reproduce the failure without the patch, and with the patch it works. So I think I will include that it 1.6.

**Comment by jcflack** [comment]
Believed resolved in 1.6.0, and backpatched to the 1.5 branch in 1.5.7.
